core/src/main/java/io/methvin/watcher/DirectoryWatcher.java

```diff
@@ -245,16 +254,21 @@ private void register(Path directory, boolean useFileTreeModifier) throws IOExce
   }
 
   private void notifyCreateEvent(Path path, int count) throws IOException {
-    HashCode newHash = PathUtils.hash(path);
-    if (newHash == null) {
-      logger.debug("Failed to hash created file [{}]. It may have been deleted.", path);
-      return;
-    }
-    // Notify for the file create if not already notified
-    if (!pathHashes.containsKey(path)) {
+    if(settings.isPreventFileHashing() && Files.isRegularFile(path)) {
```

Comment:
Why are you using `isDirectory` in the other place and `isRegularFile` here?

Also let's keep the formatting consistent and use `if (` rather than `if(`